jbpm-wb-executor-service/jbpm-wb-executor-service-client/src/main/java/org/jbpm/workbench/es/client/editors/quicknewjob/NewJobPresenter.java

```diff
@@ -0,0 +1,185 @@
+/*
+ * Copyright 2018 Red Hat, Inc. and/or its affiliates.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.jbpm.workbench.es.client.editors.quicknewjob;
+
+import java.util.Date;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import javax.annotation.PostConstruct;
+import javax.enterprise.context.Dependent;
+import javax.enterprise.event.Event;
+import javax.inject.Inject;
+
+import org.jboss.errai.bus.client.api.messaging.Message;
+import org.jboss.errai.common.client.api.Caller;
+
+import org.jbpm.workbench.es.client.i18n.Constants;
+import org.jbpm.workbench.es.model.RequestDataSetConstants;
+import org.jbpm.workbench.es.model.RequestParameterSummary;
+import org.jbpm.workbench.es.model.events.RequestChangedEvent;
+import org.jbpm.workbench.es.service.ExecutorService;
+import org.uberfire.client.mvp.UberElement;
+import org.uberfire.workbench.events.NotificationEvent;
+
+@Dependent
+public class NewJobPresenter {
+
+    private final Constants constants = Constants.INSTANCE;
+
+    private Caller<ExecutorService> executorServices;
+
+    private Event<NotificationEvent> notification;
+
+    private Event<RequestChangedEvent> requestCreatedEvent;
+
+    private String serverTemplateId;
+
+    @Inject
+    protected NewJobView view;
+
+    @PostConstruct
+    public void init() {
+        view.init(this);
+    }
+
+    public NewJobView getView() {
+        return view;
+    }
+
+    public void show(String serverTemplateId) {
```

Comment:
I can't shake of the feeling that this method should be divided in two - generic setter for _serverTemplateId_ and another method that will call only ```view.show()``` (maybe even named more descriptively e.g. ```openNewJobDialog()```).
The reason for this is that _serverTemplateId_ doesn't play any role in opening the new job dialog per se, but only later as part of calling the backend when the schedule request is sent to kie server.
However, this distinction is well hidden when the method is called in _RequestListPresenter_ and the reader is bound to assume the correlation whereas there is none.
Just a suggestion though :)

Reply:
Mixed feelings here, I get your point @bsremac, but to me is also kind of a logical dependency. You're explicitly telling to create/show a job for that server template. Thinking of the domain data that this UI depends on, like if we had to show details of a process instance in a pop up, I would either pass its id or a complete object in here.

Reply:
@cristianonicolai Right, I get it. Considering everything now, I think there are more reasons to pass it then not. Let's leave it as it is.
@nmirasch Maybe just renaming the method then to something which reflects this approach (```setNewJobDialog```?), as we are setting in the method whats needed for creating a new job eventually.